### ISSUE TITLE
Error when making plots in write_roi() method with fermipy 1.0.0 and fermitools 2.0.0

### ISSUE THREAD

**Issue by ldivenere** [fermiPy/fermipy#371]
I am using fermipy 1.0.0 and fermitools 2.0.0.
I have this error when trying to make plots in write_roi() method:
```
>>> gta.write_roi("prova.npy", make_plots=True)
2020-11-09 19:29:15 INFO    GTBinnedAnalysis.write_xml(): Writing /lustrehome/divenere/Fermi/test_fermipy/test/prova_00.xml...
2020-11-09 19:29:15 INFO    GTAnalysis.write_fits(): Writing /lustrehome/divenere/Fermi/test_fermipy/test/prova.fits...
WARNING: Format %s cannot be mapped to the accepted TDISPn keyword values.  Format will not be moved into TDISPn keyword. [astropy.io.fits.column]
WARNING: Format %f cannot be mapped to the accepted TDISPn keyword values.  Format will not be moved into TDISPn keyword. [astropy.io.fits.column]
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(138): Could not remove or rename /lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/conda-meta/numpy-1.19.2-py37h7008fea_1.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(138): Could not remove or rename /lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/conda-meta/numpy-1.19.2-py37h7008fea_1.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(138): Could not remove or rename /lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/conda-meta/numpy-1.19.2-py37h7008fea_1.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(138): Could not remove or rename /lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/conda-meta/numpy-1.19.2-py37h7008fea_1.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(138): Could not remove or rename /lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/conda-meta/numpy-1.19.2-py37h7008fea_1.json.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(138): Could not remove or rename /lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/conda-meta/numpy-1.19.2-py37h7008fea_1.json.  Please remove this file manually (you may need to reboot to free file handles)
2020-11-09 19:29:45 INFO    GTAnalysis.write_roi(): Writing /lustrehome/divenere/Fermi/test_fermipy/test/prova.npy...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/gtanalysis.py", line 3754, in write_roi
    **kwargs.get('plotting', {}))
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/gtanalysis.py", line 3829, in make_plots
    plotter.run(self, mcube_map, prefix=prefix, **kwargs)
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py", line 933, in run
    **kwargs)
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py", line 1218, in make_roi_plots
    rp = ROIPlotter(mcube_tot, roi=gta.roi, **roi_kwargs)
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py", line 393, in __init__
    self._implot = ImagePlotter(self._data_map.sum_over_axes())
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/gammapy/maps/wcsnd.py", line 226, in sum_over_axes
    return self.reduce_over_axes(func=np.add, axes=axes, keepdims=keepdims)
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/gammapy/maps/wcsnd.py", line 252, in reduce_over_axes
    map_out = map_out.reduce(ax, func=func, keepdims=keepdims)
  File "/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/gammapy/maps/wcsnd.py", line 282, in reduce
    self.data, axis=idx, keepdims=keepdims, where=~np.isnan(self.data)
TypeError: 'where' is an invalid keyword argument for reduce()
```
If I call write_roi() with option make_plots=False, I do not have any error.
This seems to be related to a gammapy issue (https://github.com/gammapy/gammapy/issues/2677).

**Comment by jeget** [comment]
In the gammapy issue that you mentioned, the solution appears to be to update numpy to 1.17 or later. Which version of numpy do you have?

**Comment by ldivenere** [comment]
I had the 1.16.0 version. I have now updated to numpy 1.17.5 and the error disappears.
Now I get several warnings related to matplotlib, but the plots are correctly produced.
```
2020-12-11 22:03:33 INFO    GTAnalysis.write_roi(): Writing /lustrehome/divenere/Fermi/Tycho/output_fermipy/prova.npy...
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py:146: MatplotlibDeprecationWarning: Passing raw data via parameters data and lut to register_cmap() is deprecated since 3.3 and will become an error two minor releases later. Instead use: register_cmap(cmap=LinearSegmentedColormap(name, data, lut))
  plt.register_cmap(name='ds9_b', data=ds9_b)
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py:301: MatplotlibDeprecationWarning: You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap("magma"))
  colormap.set_under(colormap(0))
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/astropy/visualization/wcsaxes/core.py:458: MatplotlibDeprecationWarning:
The 'inframe' parameter of draw() was deprecated in Matplotlib 3.3 and will be removed two minor releases later. Use Axes.redraw_in_frame() instead. If any parameter follows 'inframe', they should be passed as keyword, not positionally.
  super().draw(renderer, inframe=inframe)
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py:146: MatplotlibDeprecationWarning: Passing raw data via parameters data and lut to register_cmap() is deprecated since 3.3 and will become an error two minor releases later. Instead use: register_cmap(cmap=LinearSegmentedColormap(name, data, lut))
  plt.register_cmap(name='ds9_b', data=ds9_b)
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/fermipy/plotting.py:301: MatplotlibDeprecationWarning: You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap("magma"))
  colormap.set_under(colormap(0))
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/astropy/visualization/wcsaxes/core.py:458: MatplotlibDeprecationWarning:
The 'inframe' parameter of draw() was deprecated in Matplotlib 3.3 and will be removed two minor releases later. Use Axes.redraw_in_frame() instead. If any parameter follows 'inframe', they should be passed as keyword, not positionally.
  super().draw(renderer, inframe=inframe)
/lustrehome/exp_soft/glast_local/anaconda3/envs/FT_2_0_0/lib/python3.7/site-packages/astropy/visualization/wcsaxes/core.py:458: MatplotlibDeprecationWarning:
The 'inframe' parameter of draw() was deprecated in Matplotlib 3.3 and will be removed two minor releases later. Use Axes.redraw_in_frame() instead. If any parameter follows 'inframe', they should be passed as keyword, not positionally.
  super().draw(renderer, inframe=inframe)
>>>
```
If numpy version 1.17 or higher does not break other stuff, this should probably become the default.

**Comment by jeget** [comment]
I have the same warnings, too. And the plots are produced.

**Comment by eacharles** [comment]
Fixed in #380
